starfish/imagestack/imagestack.py

```diff
@@ -857,12 +857,24 @@ def _processing_workflow(
             selector_and_slice_list: Tuple[Mapping[Axes, int],
                                            Tuple[Union[int, slice], ...]],
     ):
+        # build the numpy array from the shared memory object
         backing_mp_array, shape, dtype = SharedMemory.get_payload()
         unshaped_numpy_array = np.frombuffer(backing_mp_array.get_obj(), dtype=dtype)
-        numpy_array = unshaped_numpy_array.reshape(shape)
 
-        sliced = numpy_array[selector_and_slice_list[1]]
+        # get the correct dimension order for the imagestack based on AXES_DATA
+        dims = [
+            name.value for (name, data) in
+            sorted(AXES_DATA.items(), key=lambda kv: kv[1].order)
+        ] + [Axes.Y.value, Axes.X.value]
```

Comment:
we're just trying to maintain the current axes order here right? why not just `dims = [dim for dim in self.xarray.dims]`

Reply:
Yes, but this is a static method so does not have access to self. Happy to simplify this if you can find an easier way!

Reply:
ohhhh I missed that. hmm well you could make dims_order a param you pass into _processing_workflow, then do then above in transform()... might be a little more clear